src/main/java/org/deepsymmetry/beatlink/VirtualCdj.java

```diff
@@ -1176,8 +1185,13 @@ public synchronized void stop() {
             setTempoMaster(null);
 
             if (inOpusQuadCompatibilityMode()) {
-                // We were running in proxy mode, so we need to shut down VirtualRekordbox, and nothing further.
-                VirtualRekordbox.getInstance().stop();
+                // We were running in proxy mode, so we need to shut down VirtualRekordbox (MODE 1) or OpusProvider (MODE 2), and nothing further.
```

Comment:
The “nothing further” part isn’t true any more if we were in mode 2: we need to shut down VirtualCdj itself in that mode. So we need to distinguish which mode we were in at the start here, and only `return` on line 1198 if we were in mode 1.